src/AspNetCore/Inc/application.h

```diff
@@ -181,6 +183,51 @@ class APPLICATION
         return m_pProcessManager->GetProcess( context, pConfig, ppServerProcess );
     }
 
+    HRESULT 
+    GetAspNetCoreApplication(
+        _In_    ASPNETCORE_CONFIG       *pConfig,
+        _Out_   ASPNETCORE_APPLICATION  **ppAspNetCoreApplication
+    )
+    {
+        
+        HRESULT hr = S_OK;
+        BOOL fLockTaken = FALSE;
+        ASPNETCORE_APPLICATION *application;
+
+        if (m_pAspNetCoreApplication == NULL)
+        {
+            AcquireSRWLockExclusive(&m_srwLock);
+            fLockTaken = TRUE;
+
+            if (m_pAspNetCoreApplication == NULL)
+            {
+                application = new ASPNETCORE_APPLICATION();
+                if (application == NULL) {
```

Comment:
you failed to create a new application. It should be an error while you still return S_OK as you did not set the hr value to out of memory